File: src/leetcode/2026/05/Solution908.java
public class Solution908 {
    public int smallestRangeI(int[] nums, int k) {
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;
        for (int num : nums) {
            max = Math.max(max, num);
            min = Math.min(min, num);
        }
        return Math.max(0, max - min - 2 * k);
    }

    public static void main(String[] args) {
        Solution908 solution = new Solution908();

        int[] nums1 = { 1 };
        int k1 = 0;
        int result1 = solution.smallestRangeI(nums1, k1);
        System.out.println(result1); // Output: 0

        int[] nums2 = { 0, 10 };
        int k2 = 2;
        int result2 = solution.smallestRangeI(nums2, k2);
        System.out.println(result2); // Output: 6

        int[] nums3 = { 1, 3, 6 };
        int k3 = 3;
        int result3 = solution.smallestRangeI(nums3, k3);
        System.out.println(result3); // Output: 0
    }
}

File: src/leetcode/2026/05/Solution908.cpp
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

class Solution908 {
public:
    int smallestRangeI(vector<int>& nums, int k) {
        int min_val = *min_element(nums.begin(), nums.end());
        int max_val = *max_element(nums.begin(), nums.end());
        return max(0, max_val - min_val - 2 * k);
    }
};

int main() {
    Solution908 solution;

    vector<int> nums1 = {1, 3, 6};
    int k1 = 3;
    int result1 = solution.smallestRangeI(nums1, k1);
    cout << result1 << endl; // Output: 0

    vector<int> nums2 = {0, 10};
    int k2 = 2;
    int result2 = solution.smallestRangeI(nums2, k2);
    cout << result2 << endl; // Output: 6

    vector<int> nums3 = {1};
    int k3 = 0;
    int result3 = solution.smallestRangeI(nums3, k3);
    cout << result3 << endl; // Output: 0

    return 0;
}